### ISSUE TITLE
Rename Iterators.partition

### ISSUE THREAD

**Issue by jariji** [JuliaData/DataFrames.jl#3284]
https://github.com/JuliaData/DataFrames.jl/pull/3212 adds `Iterators.partition` for `DataFrame`s.

Splitting a df into `SubDataFrame`s sounds like a useful functionality, but `Iterators.partition` is supposed to split iterables, and that is not what `DataFrame` is, so using this function is a pun. I propose `partitionrows` or something.

**Comment by bkamins** [comment]
The reason why we call it `Iterators.partition` is twofold:
* first to make it clear that we produce `Base.Iterators.PartitionIterator` object - this is relevant e.g. in Arrow.jl
* `Iterators.partition` does not require an iterator, but a collection. By convention data frame is considered a collection of rows (although this collection is not iterable nor indexable, but supports many other functions that support collections like `push!`, `append!`, `deleteat!` etc.)

In summary - given the conventions we use I prefer to keep things as they are now (otherwise we would need to rename all the functions that assume that data frame is collection of rows).

@nalimilan, @quinnj - what do you think

**Comment by quinnj** [comment]
I agree that I think it's fine as-is; it doesn't seem to me to be too bad of a pun to define partition on `DataFrame` like this; as @bkamins said, even though a DataFrame itself isn't iterable (on purpose, to avoid confusion between iterating rows or columns), it's definitely viewed as a collection, either of rows or columns.

**Comment by nalimilan** [comment]
Yes, we generally use the convention that functions operating on rows do not have the "rows" suffix (e.g. `deleteat!`), but functions operating on columns have the "columns" suffix.

**Comment by bkamins** [comment]
OK - so I am closing this issue and we leave things as they are. Thank you for discussion as it is useful to confirm the design.
